### ISSUE TITLE
Ignoring some annotations when training

### ISSUE THREAD

**Issue by dabeschte** [zylo117/Yet-Another-EfficientDet-Pytorch#687]
First of all: Thanks for the awesome code & readme!

I am trying to train on MOT16 and this dataset also provides annotations for reflections which should neither be trained nor penalized. This way, the network does not have to learn the difference between actual people and people in reflections, but it also is not penalized for detecting them.
Is there an easy way to discard these predictions / adjust their weights?

**Comment by dabeschte** [comment]
I found a way to solve this issue by setting the loss for all positive anchors with a target class that can be ignored to 0. I also saw that you first filter out all annotations with a class of -1. Does this have the same effect?

**Comment by zylo117** [comment]
what is mot16

**Comment by dabeschte** [comment]
multiple object tracking benchmark.
https://motchallenge.net/data/MOT16/

I basically need to train the network on a object tracking _video_ dataset with fixed cameras. And Mot16 does provide a subset of fixed camera videos.

Anyways, you are automatically discarding all annotations with a "-1" class. If I understand the code correctly, this just discard the target annotation, but it would still penalize the network if it detects an object at this location, because there should not be any since you discarded it first. Is there also a way to ignore such annotations completely and not penalize them?

**Comment by navegante26** [comment]
Hello @dabeschte, did you find a way to ignore some classes ? Thanks !

**Comment by dabeschte** [comment]
I honestly don't remember anymore. But I don't think so. I guess I just lowered the weight of that class. But that does not result in the same outcome
